pkgs/development/libraries/libfive/default.nix

```diff
@@ -26,8 +27,22 @@ mkDerivation {
   nativeBuildInputs = [ wrapQtAppsHook cmake ninja pkg-config ];
   buildInputs = [ eigen zlib libpng boost guile ];
 
+  # No rules to install the mac app, so do it manually.
+  postInstall = lib.optionalString stdenv.isDarwin ''
+    mkdir -p $out/Applications
+    cp -r studio/Studio.app $out/Applications/Studio.app
+    prog=$out/Applications/Studio.app/Contents/MacOS/Studio
+
+    install_name_tool \
+      -change libfive.dylib $out/lib/libfive.dylib \
+      -change libfive-guile.dylib $out/lib/libfive-guile.dylib \
+      $prog
+
+    wrapQtApp $prog
+  '';
+
   # Link "Studio" binary to "libfive-studio" to be more obvious:
-  postFixup = ''
+  postFixup = lib.optionalString (!stdenv.isDarwin) ''
     ln -s "$out/bin/Studio" "$out/bin/libfive-studio"
   '';
```

Comment:
Using if else would be clearer here.

Reply:
As in, merge both into `postInstall`?

Reply:
I meant combining postFixup and postInstall to one postInstall. Otherwise it is whatever.

Reply:
One side note: all examples I could find move comments inline in the script with this syntax. Seems ok to me.